cli/internal/hashing/package_deps_hash.go

```diff
@@ -416,3 +432,17 @@ func gitStatus(rootPath turbopath.AbsolutePath, patterns []string) (map[turbopat
 
 	return output, nil
 }
+
+// appends an item into a copy of an array to prevent mutating the original.
+// uses generics to take an array of any type and append an item of the same type.
+func copyAndAppend[K comparable](input []K, item K) []K {
+	output := make([]K, len(input))
+
+	for index, item := range input {
+		fmt.Printf("adding %#v at index %#v\n", item, index)
+		output[index] = item
+	}
+
+	output = append(output, item)
+	return output
+}
```

Comment:
We've explicitly not adopted generics yet because of https://github.com/golang/go/issues/48525. That issue is wrapping up so we can expect to upgrade Go and allow generics _soon_.

I'm super-excited to adopt them; but unfortunately, not yet.

Reply:
ahh too bad, I was excited to learn them. Backed out and will read that issue to understand it more, thanks for the pointer!